internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -729,6 +730,22 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	return reconcileComplete{dirty: true, stop: true}
 }
 
+func (r *BMACReconciler) isNonePlatform(ctx context.Context, cd *hivev1.ClusterDeployment) (isNone, requeue bool, err error) {
+	if cd.Spec.ClusterInstallRef == nil {
+		return false, false, errors.Errorf("Cluster Install Reference is null for cluster deployment ns=%s name=%s", cd.Namespace, cd.Name)
+	}
+	clusterInstall := hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: cd.Namespace,
+		Name:      cd.Spec.ClusterInstallRef.Name,
+	}
+	if err = r.Get(ctx, namespacedName, &clusterInstall); err != nil {
+		return false, !k8serrors.IsNotFound(err),
```

Comment:
if it's not found error what will re-trigger the reconcile? 
Is there some watch for cluster reference?